src/Microsoft.AspNetCore.SystemWebAdapters.Abstractions/SessionState/Serialization/JsonSessionSerializerOptions.cs

```diff
@@ -14,7 +14,7 @@ public class JsonSessionSerializerOptions
     /// <summary>
     /// Gets the mapping of known session keys to types
     /// </summary>
-    public IDictionary<string, Type> KnownKeys { get; } = new Dictionary<string, Type>();
+    public IDictionary<string, Type> KnownKeys { get; } = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This is a potentially breaking change and I'd prefer it to be done via an additional option:

```diff
public class JsonSessionSerializerOptions
{
+  public IEqualityComparer<string> KeyComparer { get; set; } = StringComparer.Ordinal;
}
```

It should also check if the `KnownKeys` has been created to throw if the setter is called. I'd also be ok if changing the comparer creates a new dictionary with the new comparer and the contents of the previous.

Reply:
If you can show that the session keys on .NET Framework were case insensitive, then we can change the default since we're already doing a major change for v2.0

Reply:
I understand. Hope this is OK.

AFAIR a lot of collections are case insensitive in .NET framework, but session for sure. This SO [comment](https://stackoverflow.com/questions/1731283/#1731535) has some alleged history. I can't find much about classic ASP anymore, but [this](https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Web/State/SessionStateItemCollection.cs#L58) is the Framework location I believe.